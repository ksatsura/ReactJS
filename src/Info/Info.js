//imports
import React from 'react';
import FilmName from '../FilmName/FilmName';
import Genre from '../Genre/Genre';
import ReleaseDate from '../ReleaseDate/ReleaseDate';
import '../style.css';

export const Info = (props) => {

    const { asset } = props;

    return (
        <div className='asset-title'>
            <div className='first-title'>
                <FilmName title={asset.title} />
                <ReleaseDate date={asset.release_date} />
            </div>
            <Genre genres={asset.genres} />
        </div>
    );
};

export default Info;