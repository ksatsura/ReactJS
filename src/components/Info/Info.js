//imports
import React from 'react';
import FilmName from '../FilmName/FilmName';
import Genre from '../Genre/Genre';
import ReleaseDate from '../ReleaseDate/ReleaseDate';
import '../../style.css';

export const Info = (props) => {

    const { film } = props;

    return (
        <div className='film-title'>
            <div className='first-title'>
                <FilmName title={film.title} />
                <ReleaseDate date={film.release_date} />
            </div>
            <Genre genres={film.genres} />
        </div>
    );
};

export default Info;