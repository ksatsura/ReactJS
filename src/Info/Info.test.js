import React from 'react';
import { Info } from './Info.js';
import { shallow } from 'enzyme';

describe('Info.js', () => {

    const props = {
        asset: {
            genres: [' ', ' '],
            release_date: ' ',
            title: '',
        }
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<Info {...props} />);

            expect(component).toMatchSnapshot();
        });

        it('renders for mobile', () => {
            const component = shallow(<Info {...props} isMobile />);

            expect(component).toMatchSnapshot();
        });
    });
});
