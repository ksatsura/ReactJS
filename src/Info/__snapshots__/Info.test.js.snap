// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info.js renders renders for desktop 1`] = `ShallowWrapper {}`;

exports[`Info.js renders renders for mobile 1`] = `ShallowWrapper {}`;
